tools/mcp/dotnet/AzureSDKDevToolsMCP/Services/GitHubService.cs

```diff
@@ -199,5 +191,41 @@ public async Task<List<string>> GetPullRequestCommentsAsync(string repoOwner, st
                 return responseList;
             }
         }
+
+        public async Task<List<String>> GetPullRequestChecks(int pullRequestNumber, string repoName, string repoOwner)
+        {
+            var checkResults = new List<string>();
+            try
+            {
+                var pr = await GetPullRequestAsync(repoOwner, repoName, pullRequestNumber);
+                if (pr == null)
+                {
+                    logger.LogError($"Pull request {pullRequestNumber} not found");
+                    throw new NotFoundException($"Pull request {pullRequestNumber} not found.", System.Net.HttpStatusCode.NotFound);
+                }
+
+                var checkResponse = await gitHubClient.Check.Run.GetAllForReference(repoOwner, repoName, pr.Head.Sha);
```

Comment:
Any idea whether this response is paginated and needs to be re-queried or not?

Reply:
It's not paged response. So it seems Octokit is handling it and responding as a list.